.github/workflows/redpanda-build.yml

```diff
@@ -12,59 +12,56 @@ on: [push, pull_request]
 
 jobs:
   build:
-    name: Build on ${{ matrix.container }}
+    name: build
     runs-on: ubuntu-20.04
-    container: ${{ matrix.container }}
-    strategy:
-      matrix:
-        container: ['ubuntu:21.04']
 
     steps:
         - name: checkout
           uses: actions/checkout@v2
 
-        - name: update container base
+        - name: freeup space
           run: |
-            export DEBIAN_FRONTEND=noninteractive
-            apt-get update -y
-            apt-get install -y git sudo pkg-config ccache
+            sudo rm -rf /usr/local/lib/android # will release about 10 GB if you don't need Android
+            sudo rm -rf /usr/share/dotnet # will release about 20GB if you don't need .NET
 
         - name: install dependencies
-          run: ./install-dependencies.sh
+          run: |
+            sudo apt update
+            sudo apt install -y ccache
+            sudo ./install-dependencies.sh
+
+        - name: ccache cache files
+          uses: actions/cache@v2
+          with:
+            path: ~/.ccache
+            key: ccache-${{ github.ref }}
```

Comment:
im not super familiar with how the cache action works, but does this key containing `github.ref` mean that there is a separate cache for each branch name? would that correspond to the PR or to `dev`?

Reply:
yeah, the idea is that there would be a separate cache for each branch, as opposed to one being created for each build. Since there's a 5GB limit (for the whole repo), and each build is ~200MB, having one cache file being pushed for each build would end up evicting ccache caches frequently.

>  would that correspond to the PR or to `dev`?

there'd be one for each